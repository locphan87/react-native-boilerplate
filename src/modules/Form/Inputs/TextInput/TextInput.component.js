// @flow
import React from 'react'
import { View, Text } from 'react-native'
import { Input } from 'react-native-elements'

import type { FieldProps } from '../../../../types'

import styles from './TextInput.component.style'

type Props = {
  name: string,
  placeholder: string,
  fieldProps: FieldProps
}

const TextInput = ({
  name,
  placeholder,
  fieldProps: { values, touched, errors, setFieldValue, setFieldTouched },
  ...rest
}: Props) => (
  <View style={styles.container}>
    <Input
      placeholder={placeholder}
      onChangeText={text => setFieldValue(name, text)}
      value={values[name]}
      onBlur={() => setFieldTouched(name)}
      containerStyle={styles.inputContainer}
      {...rest}
    />
    {touched[name] && errors[name] && <Text>{errors[name]}</Text>}
  </View>
)

export default TextInput
