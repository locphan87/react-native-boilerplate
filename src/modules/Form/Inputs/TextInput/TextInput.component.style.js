// @flow
import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    width: '100%'
  },
  inputContainer: {
    width: '100%'
  }
})
