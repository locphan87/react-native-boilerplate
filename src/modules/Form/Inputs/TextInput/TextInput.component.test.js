import React from 'react'
import { shallow } from 'enzyme'

import { testSnapshots } from '../../../../utils/test.util'

import TextInput from './TextInput.component'

const props = {
  name: 'title',
  placeholder: 'Enter your title',
  fieldProps: {
    values: {},
    touched: {},
    errors: {},
    setFieldValue: jest.fn(),
    setFieldTouched: jest.fn()
  }
}
const wrapper = shallow(<TextInput {...props} />)

describe('Form Inputs - TextInput', () => {
  testSnapshots(TextInput, [
    {
      props,
      description: 'basic render'
    },
    {
      props: {
        ...props,
        fieldProps: {
          ...props.fieldProps,
          values: { title: '' },
          touched: { title: true },
          errors: { title: 'Title is required' }
        }
      },
      description: 'should render an error'
    }
  ])

  describe('events', () => {
    const Input = wrapper.find('Input')
    test('should call setFieldTouched', () => {
      Input.props().onBlur()
      expect(props.fieldProps.setFieldTouched).toBeCalled()
    })
    test('should call setFieldValue', () => {
      Input.props().onChangeText('new text')
      expect(props.fieldProps.setFieldValue).toBeCalledWith('title', 'new text')
    })
  })
})
