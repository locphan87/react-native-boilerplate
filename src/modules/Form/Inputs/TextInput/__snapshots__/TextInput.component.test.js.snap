// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Inputs - TextInput snapshots basic render 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Input
    containerStyle={
      Object {
        "width": "100%",
      }
    }
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Enter your title"
  />
</View>
`;

exports[`Form Inputs - TextInput snapshots should render an error 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Input
    containerStyle={
      Object {
        "width": "100%",
      }
    }
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Enter your title"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Title is required
  </Text>
</View>
`;
